### DISCUSSION TITLE
Import dataset generic gps (ex. Emlid, Garmin, Trimble pathfinder)in Verde for decimation

### DISCUSSION THREAD

**Discussion by menarinis** [fatiando/community#121] [Q&A]
Hi, I'm Stefano Menarini from Italy and I work in agriculture. I discovered Verde after a research on how to decimate the points acquired with gps. I have this need for decimation. I acquire land metrics data on fields of a few hectares each time. I need to decimate my acquired positions along generally straight and parallel lines (the path of a tractor). I experimented with Verde's datasets and I understood how it works. Now I would like to use it with my gps data which is in text format (point name, latitude, longitude). I ask you can I use Green to decimate my gps points? If yes, how can I import my file with the data in Green and follow the decimation procedure? Thanks a lot in advance for the answer.

**Comment by santisoler** [comment]
Hi @menarinis. To load data in text files with Python you should take a look at Pandas. For example, with [`pandas.read_csv()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html) you can read CSV files, or you can use [`pandas.read_table()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_table.html) to read any txt file whose fields are separated by any delimiter.

Hope it helps!

**Comment by menarinis** [comment]
Thank you all. I managed to import some files without problems by transforming them into csv with comma separator and adding an extra column as you suggested. I followed the video to practice and arrived, using my data in Jupyter, at minute 32.13. Now I can't go on here. My data is already in UTM coordinates 33N and in the proj documentation I can't find information about the syntax part of Pyproj-proj ...lat_ts=data...
from which I get error. In the video it seems that the author wants to transform geographical coordinates or am I wrong? How can I move forward from here?
I also proceeded in this way with the following commands following your tutorial (https://www.fatiando.org/verde/latest/tutorials/decimation.html):
reducer = vd.BlockReduce(reduction=np.median, spacing=5 / 60) plt.figure(figsize=(7, 7))
ax = plt.axes(projection=ccrs.utm(33, southern_hemisphere=False))
ax.set_title("Locations of decimated data")
# Plot the bathymetry data locations as black dots
plt.plot(*coordinates, ".k", markersize=1, transform=crs)
but I get the error: ...AttributeError: module 'cartopy.crs' has no attribute 'utm'...
I would like to complete, without errors, the tutorial up to saving the reduced data file. Thanks in advance

**Comment by menarinis** [comment]
Hi, I managed to solve the previous problem, I finally arrived at the bottom by doing my first decimation of GPS points. I made several tests varying the settings to see the behavior of the routine. Now I realized something that I can not fix. **_On GPS tracker datasets that decimation always discards the first and last position of a dataset_**. For example, in this dataset attached I have numerical references such as index which start from 1 and reach around 1998. At the end of the decimation the routine discards the 1st and last position; _**it begins from 56 and ends a 1952**_ . Currently I have manually entered these two positions in the final file because I need them. I ask you if there is a way to preserve these 2 positions (start and end). Thanks in advance
This is the dataset
![dataset](https://github.com/fatiando/community/assets/21281413/b0447647-443e-4039-9fb3-5add2210129d)
This is the dataset after reduction. I use: 
> reducer = vd.BlockReduce(np.median, spacing=5)
> block_coords, block_name = reducer.filter(coordinates=(data.LONG, data.LAT), data=(data.Name, data.Alt))
![datset_reduct](https://github.com/fatiando/community/assets/21281413/466a482d-4912-45d7-befc-219e64e69b87)

**Comment by leouieda** [comment]
@menarinis reducing the name or index column will definitely not work. The reduction calculates the median of the values that fall inside a block. If your values are an index or the name of points, then their median doesn't mean anything. You'll probably need to through away that column and make a new index.
